### ISSUE TITLE
[Bug] ppo_loss is inf because discrete action probability is 0

### ISSUE THREAD

**Issue by P90-RushB** [ray-project/ray#21297]
### Search before asking

- [X] I searched the [issues](https://github.com/ray-project/ray/issues) and found no similar issues.

### Ray Component

RLlib

### What happened + What you expected to happen

When using ppo(torch version) in disrete action space, if training time is long enough, sometime will show:
WARNING:root:NaN or Inf found in input tensor.
the reason is that the action_kl in total_loss is inf in ppo_surrogate_loss function. To be exact, thats because the action probability
is 0 when logits is too low, so log(0) is computed. 
Even if the policy.kl_coeff is set to 0, policy.kl_coeff * action_kl (0 * tensor(inf) = nan) is also computed.
If action_kl can be removed optionally, maybe it's enough.

### Versions / Dependencies

Ray 2.0.0.dev0
python 3.7.11
os ubuntu20.04

### Reproduction script

import torch
import numpy as np

prev_act_logits = [-84.742, -7.2213, -0.00073051, -51.313, -103.49]
cur_act_logits = [-86.197, -5.939, -0.0026379, -53.358, -105.7]

ca_pre = torch.distributions.categorical.Categorical(logits=torch.tensor(prev_act_logits))
ca_cur = torch.distributions.categorical.Categorical(logits=torch.tensor(cur_act_logits))

assert ca_cur.probs[-1].item() == 0
kl = torch.distributions.kl.kl_divergence(ca_pre, ca_cur)
assert torch.distributions.kl.kl_divergence(ca_pre, ca_cur) == torch.tensor(np.inf)
print(0 * kl)

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

**Comment by sven1977** [comment]
This PR removes the entire term if "kl_coeff" is configured 0.0 for both tf and torch:

https://github.com/ray-project/ray/pull/21456

Hmm, the default for this parameter is 0.2 :/  so we may have to do more to prevent NaNs from being calculated, but the above PR is a start. Could you try and check, whether this is still an issue, even with kl_coeff==0.0 in your config? Thanks!

**Comment by sven1977** [comment]
As soon as `tf.experimental.numpy.nanmean` moves into the stable tf code (out of "experimental"), we should replace the `reduce_mean` in PPO's `reduce_mean_valid` util functions.

**Comment by gjoliver** [comment]
Assuming the fix @sven1977 made works, and we can turn kl_coeff off entirely now.
Please re-open if this is still problematic.
